content/style-injector.js

```diff
@@ -45,45 +63,39 @@ function createStyleInjector({compare, setStyleContent, onUpdate}) {
       } else if (ORDERED_TAGS.has(el.localName)) {
         return true;
       }
-      el = el.nextSibling;
+      el = el.nextElementSibling;
     }
     // some styles are not injected to the document
     return i < list.length;
   }
 
   function addMany(styles) {
-    const pending = Promise.all(styles.map(_add));
-    emitUpdate();
-    return pending;
-  }
-
-  function add(style) {
-    const pending = _add(style);
-    emitUpdate();
-    return pending;
+    const els = styles.map(_add);
+    onUpdate();
+    return els;
   }
 
   function _add(style) {
+    const code = style.code = style.code.join('');
```

Comment:
Because it had the raw `code` from the API message which is an array. Previously there was a `map()` call to convert the `code` in apply.js but I don't see the need for for these abstractions and middleware as the injector won't be used by anything else.

Reply:
Do you want to avoid creating a temporary array? Then it should be better to replace `Object.values` to something else:
```diff
diff --git a/content/apply.js b/content/apply.js
index 7b35b1fc..c70bb076 100644
--- a/content/apply.js
+++ b/content/apply.js
@@ -330,16 +330,25 @@ const APPLY = (() => {
     });
   }
 
+  function sectionsToStyles(obj) {
+    const styles = [];
+    for (const key in obj) {
+      styles.push({
+        id: obj[key].id,
+        code: obj[key].code.join('')
+      });
+    }
+    return styles;
+  }
+
   function applyStyles(sections) {
-    const styles = Object.values(sections);
+    const styles = sectionsToStyles(sections);
     if (!styles.length) {
       return Promise.resolve();
     }
     return rootReady().then(() =>
       docRootObserver.evade(() =>
-        styleInjector.addMany(
-          styles.map(s => ({id: s.id, code: s.code.join('')}))
-        )
+        styleInjector.addMany(styles)
       )
     );
   }
```
Therefore we don't have to change the signature of `styleInjector`.

Reply:
What I really want is to avoid 10 lines of intermediate unnecessary code where a single short statement does the job.

Reply:
I've switched to `styles.forEach(s => (s.code = s.code.join('')));` in applyStyles

Reply:
Ugh, fine, let's use styleMapToArray.